mlir/test/Dialect/MIOpen/lowering_bound_check.mlr

```diff
@@ -0,0 +1,22 @@
+    %8 = miopen.transform(%7) {intermediate_layout = ["go", "no", "ko", "ydot", "htilda", "xdot", "wtilda"], layout = [{dimensions = [0 : i32], names = ["go"], source_dimensions = [0 : i32], source_names = ["go"], transformation = "PassThrough"}, {dimensions = [1 : i32], names = ["no"], source_dimensions = [1 : i32], source_names = ["no"], transformation = "PassThrough"}, {dimensions = [2 : i32], names = ["ko"], source_dimensions = [2 : i32], source_names = ["ko"], transformation = "PassThrough"}, {begins = [0 : i32, 0 : i32], dimensions = [3 : i32, 5 : i32], ends = [3 : i32, 3 : i32], names = ["ydotslice", "xdotslice"], source_dimensions = [3 : i32, 5 : i32], source_names = ["ydot", "xdot"], transformation = "Slice"}, {begins = [1 : i32, 1 : i32], dimensions = [4 : i32, 6 : i32], ends = [15 : i32, 15 : i32], names = ["htildaslice", "wtildaslice"], source_dimensions = [4 : i32, 6 : i32], source_names = ["htilda", "wtilda"], transformation = "Slice"}], output_layout = ["go", "no", "ko", "ydotslice", "htildaslice", "xdotslice", "wtildaslice"]} : memref<1x32x32x3x16x3x16xf32> to memref<1x32x32x3x14x3x14xf32>
+    %9 = miopen.transform(%8) {gridwise_gemm_argument_position = 1 : i32, intermediate_layout = ["go", "no", "ko", "ydotslice", "htildaslice", "xdotslice", "wtildaslice"], layout = [{dimensions = [0 : i32], names = ["gemmG"], source_dimensions = [0 : i32], source_names = ["go"], transformation = "PassThrough"}, {dimensions = [1 : i32], names = ["gemmK"], source_dimensions = [2 : i32, 3 : i32, 5 : i32], source_names = ["ko", "ydotslice", "xdotslice"], transformation = "Merge"}, {dimensions = [2 : i32], names = ["gemmN"], source_dimensions = [1 : i32, 4 : i32, 6 : i32], source_names = ["no", "htildaslice", "wtildaslice"], transformation = "Merge"}], output_layout = ["gemmG", "gemmK", "gemmN"]} : memref<1x32x32x3x14x3x14xf32> to memref<1x288x6272xf32>
+    miopen.gridwise_gemm(%2, %9, %6) {arch = "gfx906", dilations = [1 : i32, 1 : i32], filter_dimension = [1, 32, 32, 3, 3], filter_layout = ["g", "k", "c", "y", "x"], input_dimension = [32, 1, 32, 14, 14], input_layout = ["ni", "gi", "ci", "hi", "wi"], kernel_algorithm = "backward_data_v4r1", num_cu = 64 : i32, output_dimension = [32, 1, 32, 14, 14], output_layout = ["no", "go", "ko", "ho", "wo"], padding = [[1 : i32, 1 : i32], [1 : i32, 1 : i32]], strides = [1 : i32, 1 : i32]} : memref<1x288x32xf32>, memref<1x288x6272xf32>, memref<1x32x6272xf32>
+    return
+  }
+//CHECK_AFFINE: (0, 0, 0, 1, 1)
+//CHECK_STEP3: #map0 = affine_map<(d0, d1, d2, d3, d4) -> (0, 0, 0, 1, 1)>
```

Comment:
test here is not enough. need more checks on different scenarios.

Reply:
we need tests to consider where `Embed` is used to cause dimensions which need OOB check.